src/derivative_utils.jl

```diff
@@ -645,12 +645,13 @@ end
 
 function calc_rosenbrock_differentiation!(integrator, cache, dtd1, dtgamma, repeat_step, W_transform)
   nlsolver = nothing
+  calc_tderivative!(integrator, cache, dtd1, repeat_step)
```

Comment:
I am not sure if I understand this. What different does it make if we call it earlier?